### PR TITLE
Check spotless formatting in CI

### PR DESCRIPTION
**Pull request by cmnrd**
Since we now formatted all the code with spotless, we should make sure that it stays this way. This PR re-enables the CI check that ensures that everything is correctly formatted.

### CODE REVIEW

**Review by petervdonovan — APPROVED** [review]
LGTM. As usual we should change `check-format.yml@ci-check-formatting` to `check-format.yml@master` before merging.

**Comment by lhstrh** [comment]
I recommend doing it _after_ merging. Just don't delete the branch or else CI will fail immediately.

**Review by lhstrh — APPROVED** [review]
LTGM. @petervdonovan does this mean that we can remove the formatting check from the unit tests? It would be nice if that were the case...

**Comment by petervdonovan** [comment]
I deleted the part of `RoundTripTests.java` that causes many headaches.

I did not create `FormattingUnitTests` -- I think Clement created it to try to isolate individual problems, which makes them more like unit tests, whereas the formatting tests are larger, more expensive, and harder to use for debugging. Probably we should keep `FormattingUnitTests` since they solve a different problem from the formatting check.

File: .github/workflows/check-format.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Test the automatic code formatting task
+name: Check if all files are formatted correctly with spotless
```
Comment (lhstrh) [inline .github/workflows/check-format.yml:1]:
```suggestion
name: Spotless check
```

Comment (lhstrh) [reply .github/workflows/check-format.yml:1]: Note that this is rendered in the left-hand menu, and if it is too long it just gets cut off, so it's better to keep this short.

**Review by lhstrh — APPROVED** [review]

File: .github/workflows/check-format.yml (hunk 1 above)
Comment (lhstrh) [reply .github/workflows/check-format.yml:1]: :rofl: @petervdonovan you just merged this. Could you apply this change in `master`? Thanks!

**Pull request merged**